### PR TITLE
first steps of forecast

### PR DESCRIPTION
**Pull request by sudiptoguha**
*Description of changes:* RCF already provides a forecast/extrapolate functionality based on repeated estimation of the conditional median. This PR exposes the errors in the estimation of those conditional medians. They may be helpful in. estimating confidence bounds using model assumptions extraneous to the non-parametric estimation. In the next few PRs we would enable forecasting for TRCFs as well.

### CODE REVIEW

File: Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java [hunk 1]
```diff
@@ -568,11 +569,16 @@ public float[] lastShingledPoint() {
 
     /**
      *
-     * @return the sequence index of the last known shingled point
+     * @return the sequence index of the last known shingled point if internal
```
Comment (jotok) [inline Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java:572]: If shingle-size is 10 and internal shingling is disabled, then `nextSequenceIndex` should always return 9?

File: Java/core/src/main/java/com/amazon/randomcutforest/imputation/ImputeVisitor.java [hunk 2]
```diff
@@ -174,6 +182,13 @@ public void acceptLeaf(final INodeView leafNode, final int depthOfNode) {
             }
         }
 
+        if (centrality < 1.0) {
+            Random rng = new Random(randomSeed);
+            randomSeed = rng.nextLong();
+            rng.nextDouble(); // discard
```
Comment (jotok) [inline Java/core/src/main/java/com/amazon/randomcutforest/imputation/ImputeVisitor.java:188]: Why?

**Review by jotok — APPROVED** [review]

File: Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java (hunk 1 above)
Comment (sudiptoguha) [reply Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java]: Fixed. I was confused/forgot what the function was doing.

File: Java/core/src/main/java/com/amazon/randomcutforest/imputation/ImputeVisitor.java (hunk 2 above)
Comment (sudiptoguha) [reply Java/core/src/main/java/com/amazon/randomcutforest/imputation/ImputeVisitor.java]: I was attempting to decor relate a linear congruential pseudo random generator. But really this cannot be solved in this library. Removing.

**Review by jotok — APPROVED** [review]

**Pull request merged**